### PR TITLE
[Testing:Submission] Notebook Builder And Panel Tests

### LINKED ISSUE #10758: Uncaught SyntaxError: Identifier has already been declared
**Describe the bug**
During testing for notebook view panel , when i add the notebook from notebook builder and move back to ta grading interface i am getting two uncaught syntaxerror
```js
Uncaught SyntaxError: Identifier 'showPopup' has already been declared
Uncaught SyntaxError: Identifier 'colorEvent' has already been declared
```
For `showPopup ` it is due to naming conflict, as we have variable and function with the same name 
For `colorEvent ` i did not get the possible reason for the syntax error

**Expected behavior**
We have to fix the syntax error , so cypress test does not fail due to this syntaxError

**To Reproduce**
1. Submitty > sample > Edit Gradeable(http://localhost:1511/courses/f24/sample/gradeable/grading_homework/update?nav_tab=1)
2. From notebook builder click on start new and create a notebook 
3. Go back to rubric grading interface (http://localhost:1511/courses/f24/sample/gradeable/grading_homework/grading/grade?who_id=jKjodWaRdEV9pBb&sort=id&direction=ASC) 
4. See the error from inspect
5. 
**Configuration**
 - OS: [Windows]
 - Browser [Chrome]

**Screenshots**
`Before the notebook `

before_the_notebook

`After adding the notebook from notebook builder `
when_we_add_notebook_from_notebook_builder

**Additional context**

### PR DESCRIPTION
**Pull request by manzil-infinity180**
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant
* [ ] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?

* No test present for notebook sections

### What is the new behavior?
So let me explain what i am doing in this pr
* Good to merge this pr after this issue got resolved #10758
* Added the test for notebook section (notebook builder section and notebook view section)
* Add the test for `start new `and `edit existing `button 
Screenshot 2024-08-01 at 10 53 56 AM

* Later added assertion for the `markdown`, `multiple choice`, and other present buttons
<br>
Screenshot 2024-08-01 at 10 54 20 AM

* Added simple text from markdown , and also verified (assertion) for the same text from ta grading section( notebook view) and then remove the notebook or we can say the added text from the widget control (remove button)
Screenshot 2024-08-01 at 10 55 01 AM

### CODE REVIEW

File: site/cypress/e2e/Cypress-TAGrading/notebook_section.spec.js
Comment (IDzyre) [inline site/cypress/e2e/Cypress-TAGrading/notebook_section.spec.js:3]: Would it be better to fix the JS exceptions rather than catching and ignoring them?

Comment (manzil-infinity180) [reply site/cypress/e2e/Cypress-TAGrading/notebook_section.spec.js:3]: Ok i will do it

**Comment by williamjallen** [comment]
@manzil-infinity180 It looks like there is currently a conflict which needs to be resolved.

File: site/app/templates/notebook/Notebook.twig
Comment (williamjallen) [inline site/app/templates/notebook/Notebook.twig:293]: Why was this change necessary?  If you're fixing a bug these tests uncovered, would it be better to make a separate bugfix PR and then update this PR once that gets merged?

File: site/cypress/e2e/Cypress-TAGrading/notebook_section.spec.js
Comment (williamjallen) [inline site/cypress/e2e/Cypress-TAGrading/notebook_section.spec.js:6]: Can you leave some comments explaining what each section of the test is supposed to test?

**Review by williamjallen — DISMISSED** [review]

File: site/app/templates/notebook/Notebook.twig
Comment (manzil-infinity180) [reply site/app/templates/notebook/Notebook.twig]: reverted the changes

File: site/cypress/e2e/Cypress-TAGrading/notebook_section.spec.js
Comment (manzil-infinity180) [reply site/cypress/e2e/Cypress-TAGrading/notebook_section.spec.js:2]: Done [fcc6046](https://github.com/Submitty/Submitty/pull/10766/commits/fcc6046db5f0a273a8881bb7addc2d1dfeb1bd5a)

**Comment by IDzyre** [comment]
It looks like a few of the tests are failing now, if you get them to pass then I will do a final review and merge.

**Comment by manzil-infinity180** [comment]
> It looks like a few of the tests are failing now, if you get them to pass then I will do a final review and merge.



> It looks like a few of the tests are failing now, if you get them to pass then I will do a final review and merge.

https://github.com/Submitty/Submitty/pull/10787 waiting for this pr to merge . It will start passing the test

**Comment by IDzyre** [comment]
If these merge tests pass, I will approve and merge

**Review by IDzyre — APPROVED** [review]

**Pull request merged**
